app/src/main/java/org/systers/mentorship/view/fragments/ChangePasswordFragment.kt

```diff
@@ -71,12 +60,27 @@ class ChangePasswordFragment : DialogFragment() {
             changePasswordView.tilNewPassword?.error = null
 
             if (validatePassword()) {
-                changePasswordViewModel.changeUserPassword(ChangePassword(currentPassword, newPassword))
+                changePassword()
             }
         }
     }
 
-    private fun validatePassword() : Boolean {
+    private fun changePassword() {
+        changePasswordViewModel.changeUserPassword(ChangePassword(currentPassword, newPassword))
+            .observe(this, Observer<Boolean> {
+                if (it != null) {
```

Comment:
Maybe adding `? ` before `.observe` could remove checking it against null?

Reply:
I found it easier to read, it's more explicit.
After all, it's almost purely stylistic.

I think we should estabilish some style guide for this project.

Reply:
The only thing is if this is null and you call method on it then the app will crash. I think its good to follow Kotlin syntax. Unless I'm wrong and changeUserPassword never return null.

Reply:
I'm not sure if you mean. `changeUserPassword()` return `LiveData<Boolean>`, so it won't be null itself. Boolean may be null but we're already checking it.